javascript/exosecurity/features/support/world.js

```diff
@@ -0,0 +1,46 @@
+const ObservableProcess = require('observable-process')
+const path = require('path')
+const { any } = require('prelude-ls')
+const { waitUntil } = require('wait')
+
+const observableProcessOptions = process.env.DEBUG_EXOSERVICE
+  ? {
+      verbose: true,
+      stdout: process.stdout,
+      stderr: process.stderr,
+    }
+  : {
+      verbose: false,
+      stdout: false,
+      stderr: false,
+    }
+
+function World() {
+  this.createExosecurityInstance = function({ role, exocomPort }, done) {
+    const fixturePath = path.join(process.cwd(), 'features', 'fixtures', role)
+    this.process = new ObservableProcess(`node ${fixturePath}`, {
+      env: {
+        EXOCOM_PORT: exocomPort,
+        ROLE: role,
+      },
+      verbose: observableProcessOptions.verbose,
+      stdout: observableProcessOptions.stdout,
+      stderr: observableProcessOptions.stderr,
+    })
+    this.process.wait('online at port', done)
+  }
+  this.removeRegisterServiceMessage = function(exocom, done) {
+    this.exocom = exocom
+    return waitUntil(() => this.exocom.receivedMessages.length, 10, () => {
```

Comment:
The return is unneeded as what the function returns is unimportant. It is passed a callback instead